File: LeetCode/Array/973.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    vector<vector<int>> kClosest(vector<vector<int>> &points, int k)
    {
        int n = points.size();
        vector<pair<int, int>> vt;

        for (int i = 0; i < n; ++i)
        {
            auto point = points[i];
            vt.push_back({ point[0] * point[0] + point[1] * point[1], i });
        }

        sort(vt.begin(), vt.end(), [](const pair<int, int> &u, const pair<int, int> &v) { return u.first < v.first; });

        vector<vector<int>> answer;

        for (int i = 0; i < k; ++i)
        {
            answer.push_back(points[vt[i].second]);
        }
        return answer;
    }
};

File: LeetCode/Math/973.py
class Solution:
    def kClosest(self, points: list[list[int]], k: int) -> list[list[int]]:
        arr = sorted([(p[0] ** 2 + p[1] ** 2, i) for i, p in enumerate(points)])[:k]
        answer = []
        for _, i in arr:
            answer.append(points[i])
        return answer
